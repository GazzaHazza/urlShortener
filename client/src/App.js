import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
   // Fetch passwords after first mount
  componentDidMount() {
    this.getPasswords();
  }

  getPasswords = () => {
    // Get the passwords and store them in state
    fetch('/api/feofo')
      .then(res => res.json())
  }
  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to React</h2>
        </div>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

export default App;
