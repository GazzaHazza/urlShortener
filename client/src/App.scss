@import './styles/_colors.scss';
body {
  background: $color5;
  color: $color3;
}
.container {
  max-width: 1080px;
  margin: 0 auto;
  display: flex;
}