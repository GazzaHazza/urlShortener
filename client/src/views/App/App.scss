@import './styles/_colors.scss';
@import './styles/_typography.scss';
body {
  background: $color5-darker;
  color: $color3-darker;
  font-family: $body-font;
  font-size: $font-size-normal;
}
.container {
  max-width: 1080px;
  margin: 200px auto 0;
  display: flex;
  flex-flow: column;
  align-content: center;
  @media all and (max-width: 520px) {
    margin-top: 50px;
  }
}