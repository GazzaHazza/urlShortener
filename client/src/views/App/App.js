import React, { Component } from "react";
import { Provider } from "react-redux";
import { BrowserRouter, Route } from "react-router-dom";
import "./App.css";
import URLRedirect from "../URLRedirect/URLRedirect";
import Home from "../Home/Home";
import store from "../../store";

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <BrowserRouter>
          <div className="container">
            <Route exact path="/" component={Home} />
            <Route path="/:urlCode" component={URLRedirect} />
          </div>
        </BrowserRouter>
      </Provider>
    );
  }
}
export default App;
