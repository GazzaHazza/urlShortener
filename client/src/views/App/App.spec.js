import React from "react";
import { shallow } from "enzyme";
import App from "./App";
import "jest-styled-components";

it("renders correctly", () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});
