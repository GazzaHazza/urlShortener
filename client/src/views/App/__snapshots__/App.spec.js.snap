// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <styled.div>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/:urlCode"
      />
    </styled.div>
  </BrowserRouter>
</Provider>
`;
